pkg/ccl/backupccl/backup_test.go

```diff
@@ -9854,3 +9854,37 @@ func TestBackupRestoreSeparateExplicitIsDefault(t *testing.T) {
 		sqlDB.Exec(t, "DROP DATABASE inc_fkdb;")
 	}
 }
+
+// TestOnFailOrCancelExecCfgNPE is a regression test that ensures the execCfg on
```

Comment:
nit: I'd say it is more broad than just checking execCfg; it is checking that we cancel successfully after pausing mid-backup, which just happens to include this case, but covers others too, right?